src/main/java/org/betonquest/betonquest/quest/event/legacy/QuestEventAdapter.java

```diff
@@ -39,6 +39,7 @@ public QuestEventAdapter(final Instruction instruction, final Event event, final
         this.event = event;
         this.staticEvent = staticEvent;
         staticness = staticEvent != null;
+        persistent = true;
```

Comment:
I do not understand this change :/

Reply:
This was a bug in the initial implementation. The message of the commit that did this change is:
> Make events using QuestEventAdapter persistent by default.
>
> To migrate events that where not persistent an appropriate decorator can
be used.

And the code in the (legacy) QuestEvent class that handles persistence looks like this:
```java
    private void handleOfflineProfile(final Profile profile) throws QuestRuntimeException {
        if (persistent) {
            LOG.debug(instruction.getPackage(), "Persistent event will be fired for offline profile.");
            handle(profile);
        } else {
            LOG.debug(instruction.getPackage(), "Player " + profile.getPlayer() + " is offline, cannot fire event because it's not persistent.");
        }
    }
```

Thus, when not doing this change it is impossible to adapt persistent events correctly. The `OnlinePlayerRequired` decorator enables events that are not persistent to preserve this behavior.